File: array/1460_make_two_arrays_equal_by_reversing_sub_arrays/1460_make_two_arrays_equal_by_reversing_sub_arrays.go
package main

import "fmt"

func canBeEqual(target []int, arr []int) bool {
	tmp := make(map[int]int)
	for _, v := range target {
		tmp[v]++
	}
	for _, v := range arr {
		if _, ok := tmp[v]; ok {
			tmp[v]--
		} else {
			return false
		}
		if tmp[v] == 0 {
			delete(tmp, v)
		}
	}
	return len(tmp) == 0
}

func main() {
	target := []int{1, 2, 3, 4}
	arr := []int{2, 4, 1, 3}
	res := canBeEqual(target, arr)
	fmt.Println(res)
}

File: array/1460_make_two_arrays_equal_by_reversing_sub_arrays/1460_make_two_arrays_equal_by_reversing_sub_arrays.py
import collections

from typing import List


class Solution:
    
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        c = collections.Counter(target)
        for a in arr:
            c[a] -= 1
            if c[a] < 0:
                return False
        return True    


target = [1, 2, 3, 4]
arr = [2, 4, 1, 3]
res = Solution().canBeEqual(target, arr)
print(res)